### ISSUE TITLE
All non-default imports from react-navigation are undefined

### ISSUE THREAD

**Issue by Un3qual** [react-community/react-navigation#1631]
### Current Behavior

![image](https://cloud.githubusercontent.com/assets/6117075/26289333/feadb2c6-3e53-11e7-8465-d4c7298f2b12.png)
`(0 , _reactNavigation.TabNavigator) is not a function`

### Expected Behavior

I don't get that error

### Your Environment

```
import { addNavigationHelpers, TabNavigator, StackNavigator, TabBarTop } from 'react-navigation'
...
const HomeNavigator = TabNavigator({
...
  Settings: {
    screen: SettingsScreen,
    path: 'settings',
  },
...
})

const AppWithNavigationState = ({ dispatch, nav }) => (
  <HomeNavigator navigation={addNavigationHelpers({ dispatch, state: nav })} />
)
```
If I change `TabNavigator` to `StackNavigator`, I get the same error but it's `(0 , _reactNavigation.StackNavigator) is not a function`\

If I `console.log` TabNavigator, it's undefined. Am I just importing it wrong?

For some reason, if I disable remote JS debugging, the error changes to 
```
 ( is not a function. (In '(0, _reactNavigation.TabNavigator)', '(' is undefined)
```

### All non default imports are undefined. 
if I `import something from 'react-navigation'` I can get an object with all of the methods, but they're all undefined: 
![image](https://cloud.githubusercontent.com/assets/6117075/26290877/88b26538-3e5f-11e7-96fb-d756c0285b9f.png)

I've tried using multiple different versions of `react-navigation`.
I tried running a backed up version from a few days ago that worked at the time. It has also stopped working with the same error. 😭

| software         | version
| ---------------- | -------
| react-navigation |  1.0.0-beta.9
| react-native     |  0.44.0
| node             | 7.7.1
| npm or yarn      | yarn 0.21.3

**Comment by jeloagnasin** [comment]
+1

**Comment by jeloagnasin** [comment]
Hey @Un3qual,

I encountered the same issue and was able to fix it (after roughly 6 hours of hair pulling).

The issue with my project is that I added `babel-plugin-add-module-exports`. And it somehow messes with the transpilation of the code. I'm not sure why or how. I removed the plugin from my `package.json` and `.babelrc`. Then, I cleaned my project and build.

To clean my project, I usually do the following commands (not sure what works but this is what I've been reading online):

```bash
$ watchman watch-del-all
$ rm -rf ~/Library/Developer/Xcode/DerivedData
$ rm -rf node_modules/
$ rm -rf ios/build/
$ rm -rf android/build/
$ rm -rf $TMPDIR/react-*
$ rm yarn.lock
$ yarn cache clean
```

And to build, I do:

```bash
$ cd ios && pod install && cd ..
$ yarn
$ react-native link
$ react-native run-ios
```

I hope this helps you.
